### PR TITLE
Docs: Remove Elastic references from .md docs

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

- update FAQ to narrow scope, clarify semver, remove why opensource question
- reomove reference to Elastic Container Registry
- remove links to starter projects
- update doc site links
- remove backport section from release process doc and add legacy notice (Note: I left this file for now, as it's a useful reference for what release automation exists that we may want to re-use or re-implement)
 
### Issues Resolved

fixes https://github.com/opensearch-project/oui/issues/86
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] All tests pass
  - [ ] `yarn lint`
  - [ ] `yarn test-unit`
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by seanneumann — APPROVED** [review]
LGTM

**Comment by BSFishy** [comment]
Could the `releasing-versions.md` and the `RELEASING.md` files be combined in some way or is that outside of the scope of this PR?

**Review by ashwin-pc — APPROVED** [review]

**Comment by joshuarrrr** [comment]
> Could the `releasing-versions.md` and the `RELEASING.md` files be combined in some way or is that outside of the scope of this PR?

Yeah - I think the right way to go is to remove `releasing-versions.md` in favor of `RELEASING.md`. But I left it for now, because it's interesting to feature compare how the previous system worked with how we plan to do releases.

File: wiki/releasing-versions.md [hunk 1]
```diff
@@ -14,62 +16,10 @@ After the version is bumped, the release script automatically updates `CHANGELOG
...
 The command will prompt you for your git credentials. If you are using 2FA for git (which you should be) then your git password must be a [one time token](https://github.com/settings/tokens).
 
-That's it. The latest changes were published to GitHub, a new `git` tag now exists on GitHub, the new release can be installed from `npm`, and the [documentation site][docs] will update momentarily<sup>\*</sup>.
-
-<sup>_\* GitHub Pages sites are cached aggressively and can sometimes take a couple of minutes to update._</sup>
+That's it. The latest changes were published to GitHub, a new `git` tag now exists on GitHub, the new release can be installed from `npm`, and the [documentation site][docs] will update momentarily.
 
 ## `@elastic/eslint-plugin-eui`
 
 For information on releasing the eslint plugin checkout the readme in [packages/eslint-plugin/README.md](../packages/eslint-plugin/README.md)
```
Comment (BSFishy) [inline wiki/releasing-versions.md:23]: Should this be removed?

Comment (joshuarrrr) [reply wiki/releasing-versions.md:23]: yeah, it should - even if we still consume it, I don't think we have any current plans to fork or release. @AMoo-Miki is this something you already have a plan for?

**Pull request merged**